src/vos/vos_dtx.c

```diff
@@ -31,8 +31,41 @@
 #include "vos_layout.h"
 #include "vos_internal.h"
 
-#define DTX_UMMID_ABORTED	((umem_id_t){ .pool_uuid_lo = -1, .off = -1, })
-#define DTX_UMMID_UNKNOWN	((umem_id_t){ .pool_uuid_lo = -1, .off = -2, })
+static inline bool
+dtx_is_null(umem_off_t umoff)
+{
+	/* If @umoff is zero, then it will be regared as NULL case.
+	 * That is compatible with the case of NOT set DTX things.
+	 *
+	 * In theory, zero is valid offset in a PMEM pool.
+	 * UMMID_IS_NULL() also handles zero offset as NULL.
+	 */
+	return umoff == UMOFF_NULL || umoff == 0;
+}
+
+static inline bool
+dtx_is_aborted(umem_off_t umoff)
+{
+	return umem_off_get_invalid_flags(umoff) == UMOFF_ABORTED;
+}
+
+static inline bool
+dtx_is_unknown(umem_off_t umoff)
+{
+	return umem_off_get_invalid_flags(umoff) == UMOFF_UNKNOWN;
+}
+
+static void
+dtx_set_aborted(umem_off_t *umoff)
+{
+	umem_off_set_invalid(umoff, UMOFF_ABORTED);
```

Comment:
I don't think it should be UMOFF flag but rather DTX flag here though I guess there could be conflict between users if you don't reserve bits.   Just seems a bit weird to have ABORTED associated with a umem pointer.   If we do that, we might want to consider losing this API and just have an api to set something as aborted/null/unknown and call it a day.

Reply:
I will move this part into DTX logic, not the general umem.

Reply:
I think it is OK to put it in common code, this is part of DAOS anyway, if there is any chance to get conflict, then it's going to be very hard to debug.